docs/design-docs/cli/cli.md

```diff
@@ -0,0 +1,119 @@
+# Epiphany CLI design document
+
+Affected version: 0.2.1
+
+## Goals
+
+Provide simple in use cli that will:
+
+1. provide input validation (cmd arguments and data file)
+2. maintain Epiphany cluster state (json file, binary, tbd)
+3. allow to create empty project (via command-line and data file)
+4. maintain information about Epiphany version used on each machine (unique identifier generation?)
+5. allow to add/remove resources via data file.
+    - separate infrastructure data files from configuration
+    - internal file with default values will be created
+6. allow to add resources via command-line (networks, vpn, servers, roles, etc.)
+7. allow all messages from cli to be convertible to json/yaml (like -o yaml, -o json)
+8. plugable storage/vault for Epiphany state and Terraform state
+
+## Use cases
+
+### CLI deployments/management usage
+
+Create empty cluster:
+
+```bash
+> epiphany create cluster --name='epiphany-first-cluster'
```

Comment:
Wouldn't it be better to use something shorter or more meaningful(eg epicli, epictl)?

Reply:
This is just a proposition - but `epictl` sounds great

Reply:
Go for epictl. :) 👍

Reply:
my vote for epicli since that can be said as epicly